pkg/converter/target/externalentity.go

```diff
@@ -66,20 +70,65 @@ func NewExternalEntityFrom(
 
 // PatchExternalEntityFrom generate a patch for existing ExternalEntity from source.
 func PatchExternalEntityFrom(
-	source ExternalEntitySource, patch *antreatypes.ExternalEntity, cl client.Client) *antreatypes.ExternalEntity {
-	populateExternalEntityFrom(source, patch, cl)
-	return patch
+	source ExternalEntitySource, patch *antreatypes.ExternalEntity, cl client.Client) (*antreatypes.ExternalEntity, bool) {
+	changed := populateExternalEntityFrom(source, patch, cl)
+	return patch, changed
 }
 
 func populateExternalEntityFrom(source ExternalEntitySource, externalEntity *antreatypes.ExternalEntity,
-	cl client.Client) {
-	externalEntity.SetLabels(genTargetEntityLabels(source, cl))
+	cl client.Client) bool {
+	changed := false
+	if !reflect.DeepEqual(externalEntity.GetLabels(), genTargetEntityLabels(source, cl)) {
+		externalEntity.SetLabels(genTargetEntityLabels(source, cl))
+		changed = true
+	}
 	ipAddrs, _ := source.GetEndPointAddresses()
 	endpoints := make([]antreatypes.Endpoint, 0, len(ipAddrs))
 	for _, ip := range ipAddrs {
 		endpoints = append(endpoints, antreatypes.Endpoint{IP: ip})
 	}
-	externalEntity.Spec.Endpoints = endpoints
-	externalEntity.Spec.Ports = source.GetEndPointPort(cl)
-	externalEntity.Spec.ExternalNode = source.GetExternalNodeName(cl)
+	if externalEntity.Spec.ExternalNode != source.GetExternalNodeName(cl) {
+		externalEntity.Spec.ExternalNode = source.GetExternalNodeName(cl)
+		changed = true
+	}
+	sourcePorts := source.GetEndPointPort(cl)
+	sort.Slice(externalEntity.Spec.Ports, func(i, j int) bool {
+		if externalEntity.Spec.Ports[i].Name != externalEntity.Spec.Ports[j].Name {
```

Comment:
I see that for externalEntity.Spec.Ports (L96 - 102) and sourcePorts (L105-111), the comparator function is same. 
Can you define a function for the comparator and pass the comparator function to sort.Slice?